src/app/pages/private/shop-approval/shop-approval.component.ts

```diff
@@ -0,0 +1,71 @@
+import {Component, OnInit, TemplateRef, ViewChild} from '@angular/core';
+import {FormBuilder, FormGroup, Validators} from "@angular/forms";
+import {NgbModal} from "@ng-bootstrap/ng-bootstrap";
+import {ExtendedFormControl} from "../../../core/utils/extended-form-control.utils";
+import {APP_ROUTES} from "../../../core/utils/constants.utils";
+import {ActivatedRoute, Router} from "@angular/router";
+import {PlatformShopService} from "../../../core/service/platform-shop.service";
+import {ToastrService} from "ngx-toastr";
+
+@Component({
+  selector: 'shop-approval',
+  templateUrl: './shop-approval.component.html',
+  styleUrls: ['./shop-approval.component.css']
+})
+export class ShopApprovalComponent implements OnInit {
```

Comment:
`implements OnInit` should be replaced with `extends BaseComponent` in all your code changes